infra/modules/aws-cognito-identity-cli/main.tf

```diff
@@ -0,0 +1,27 @@
+provider "aws" {
+  assume_role {
+    role_arn = var.aws_role
+  }
+}
+
+resource "null_resource" "identity-command" {
+  for_each = var.login-ids
+
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+  provisioner "local-exec" {
+    command = "bash ${path.module}/addIdentity.sh ${var.identity_pool_id} ${each.value} ${var.aws_region} ${var.aws_role} ${each.key} ${path.module}"
+  }
+}
+
+data "local_file" "identity-result" {
+  for_each = var.login-ids
+
+  filename   = "${path.module}/cognito_identity_${each.key}.json"
+  depends_on = [null_resource.identity-command]
+}
+
+output "identity_id" {
```

Comment:
So the approach here is the following:
- Invoking the shell as a resource; not as a data. This is to ensure that the script is executed as part of *apply* and not as part of *plan*; it will also ensure I can have a better handling of results. Once TF supports this, we could use the proper command.
- The output of the file is dropped as a file, needed to have the *IdentityId* from JSON response there. We don't need the file for more things and as it includes a *token* we could use for authentication it is marked as ignored in *.gitignore.